actions/actions_whitebox_test.go

```diff
@@ -0,0 +1,180 @@
+		require.Equal(t, fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns], changes[0].NewValue)
+	})
+
+	s.T().Run("multiple changes", func(t *testing.T) {
+		wiCopy := createWICopy(*fxt.WorkItems[0], workitem.SystemStateOpen, []interface{}{"bcid0"})
+		fxt.WorkItems[0].Fields[workitem.SystemState] = workitem.SystemStateNew
+		fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns] = []interface{}{"bcid0", "bcid1"}
+		changes, err := fxt.WorkItems[0].ChangeSet(wiCopy)
+		require.NoError(t, err)
+		require.Len(t, changes, 2)
+		// we intentionally test the order here as the code under test needs
+		// to be expanded later, supporting more changes and this is an
+		// integrity test on the current impl.
+		require.Equal(t, workitem.SystemState, changes[0].AttributeName)
+		require.Equal(t, workitem.SystemStateNew, changes[0].NewValue)
+		require.Equal(t, workitem.SystemStateOpen, changes[0].OldValue)
+		require.Equal(t, workitem.SystemBoardcolumns, changes[1].AttributeName)
+		require.Equal(t, fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns], changes[1].NewValue)
+		require.Equal(t, wiCopy.Fields[workitem.SystemBoardcolumns], changes[1].OldValue)
+	})
+
+	s.T().Run("new instance", func(t *testing.T) {
+		fxt.WorkItems[0].Fields[workitem.SystemState] = workitem.SystemStateNew
+		fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns] = []interface{}{}
+		changes, err := fxt.WorkItems[0].ChangeSet(nil)
+		require.NoError(t, err)
+		require.Len(t, changes, 1)
+		require.Equal(t, workitem.SystemState, changes[0].AttributeName)
+		require.Equal(t, workitem.SystemStateNew, changes[0].NewValue)
+		require.Nil(t, changes[0].OldValue)
+	})
+}
+
+func (s *ActionSuite) TestActionExecution() {
+	fxt := tf.NewTestFixture(s.T(), s.DB, tf.CreateWorkItemEnvironment(), tf.WorkItems(2))
+	userID := fxt.Identities[0].ID
+
+	s.T().Run("by Old New", func(t *testing.T) {
+		fxt.WorkItems[0].Fields[workitem.SystemState] = workitem.SystemStateNew
+		fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns] = []interface{}{"bcid0", "bcid1"}
+		newVersion := createWICopy(*fxt.WorkItems[0], workitem.SystemStateOpen, []interface{}{"bcid0", "bcid1"})
+		_, changes, err := ExecuteActionsByOldNew(s.Ctx, s.GormDB, userID, fxt.WorkItems[0], newVersion, map[string]string{
+			"Nil": "{ noConfig: 'none' }",
+		})
+		require.NoError(t, err)
+		require.Len(t, changes, 0)
+	})
+
+	s.T().Run("by ChangeSet", func(t *testing.T) {
+		fxt.WorkItems[0].Fields[workitem.SystemState] = workitem.SystemStateNew
+		fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns] = []interface{}{"bcid0", "bcid1"}
+		newVersion := createWICopy(*fxt.WorkItems[0], workitem.SystemStateOpen, []interface{}{"bcid0", "bcid1"})
+		contextChanges, err := fxt.WorkItems[0].ChangeSet(newVersion)
+		require.NoError(t, err)
+		afterActionWI, changes, err := ExecuteActionsByChangeset(s.Ctx, s.GormDB, userID, newVersion, contextChanges, map[string]string{
+			"Nil": "{ noConfig: 'none' }",
```

Comment:
`{ noConfig: 'none' }` is not valid JSON but shouldn't it be JSON? I mean if `noConfig` is an unknown key, then I would understand it but here the configuration is simply invalid but no error seems to occur.

Reply:
Thank you for that explanation again and meanwhile I get it. But here it looked like it was a mistake in test, so I was confused.